### ISSUE TITLE
Unable to push - map is not a function

### ISSUE THREAD

**Issue by idanlo** [aws-amplify/amplify-cli#2365]
**Describe the bug**
Randomly when pushing I get this error.
This happened while the push was already in progress and caused us to delete the entire stack because it had too many errors which made us lose about 20 users in our user pool, all our database data (which was backed up), all AppSync custom functions and all S3 files
```
(proj.inclues || []).map is not a function
An error occured during the push operation: (proj.includes || []).map is not a function
``` 
(by the way it is spelled occurred and not occured)

**To Reproduce**
1. amplify push

**Expected behavior**
Should push changes to Amplify

**Additional context**
This happens on version 3.9.0, we downgraded to 3.0.0 and pushing works

**Comment by attilah** [comment]
@idanlo it is strange, since this issue is a duplicate of https://github.com/aws-amplify/amplify-cli/issues/2262 which was fixed back then.

Could you please share your yaml file in the root which has the structure you can see at the mentioned issue?

**Comment by idanlo** [comment]
you are correct this was the issue. the path in the `includes` property was not correct.
Still, this error message needs to change to something else that will point the user to the yaml file

**Comment by mooser** [comment]
I am also having this issue...   I was originally getting the old "stackUpdate" error, but then I deleted my api and my functions that depend on the api, recreated them, ran amplify push:

> ✖ An error occurred when pushing the resources to the cloud
> 
> (proj.includes || []).map is not a function
> An error occured during the push operation: (proj.includes || []).map is not a function

Which yml file should I be checking in?

This means I can't reliably get my project back up and running if I run into a stackupdate error while pushing.

**Comment by mooser** [comment]
You know what?  This has nothing to do with yml files in my case...  

This issue is fixed for me in PR: #2442
